plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -212,22 +210,68 @@ private void returnFailure(BulkByScrollResponse response, String modelId, Action
         actionListener.onFailure(new OpenSearchStatusException(errorMessage, RestStatus.INTERNAL_SERVER_ERROR));
     }
 
-    private void deleteModel(String modelId, ActionListener<DeleteResponse> actionListener) {
-        DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId);
-        client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
+    private void deleteModel(String modelId, FunctionName functionName, ActionListener<DeleteResponse> actionListener) {
+        // Always delete model chunks and model controller first, because deleting metadata first user is not able clean up model chunks and
```

Comment:
This could have such issue: model content deleted, but model meta still there. User can still see this model with search API. Then user may want to deploy this model again. It will throw error which may confuse user. 

That's why we want to delete model meta first. If model chunk deleted failed, we throw error and user need to re-delete the model, that's why we catch the not found exception when delete model meta and re-delete the model chunk in original code.

Reply:
If we delete model metadata first and model chunk or controller are not deleted, then there's no chance user can delete these, since the getRequest won't query results back: https://github.com/opensearch-project/ml-commons/blob/c53d586e50d7d38ccd4d5448c09bc015cce49992/plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java#L104C24-L104C34. 
For deploy model, the model query also exclude the model content: https://github.com/opensearch-project/ml-commons/blob/c53d586e50d7d38ccd4d5448c09bc015cce49992/plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java#L1444, I think it should be fine, but even if this has problem, I think we'd better to add message in deploy API.